src/CacheEntryPool/CacheEntryPool.php

```diff
@@ -228,17 +331,32 @@ public function saveDeferred(CacheItemInterface $item): bool
      */
     public function commit(): bool
     {
-        $deferred = [];
-        foreach ($this->deferredItems as $item) {
-            $deferred[$item->getKey()] = $item->get();
+        if (!count($this->deferredItems)) {
+
+            return true;
+        }
+
+        foreach ($this->deferredItems as $key => $item) {
+            if (!$this->isExpired($item)) {
+                $deferred[$key] = $item->get();
+            } else {
+                // clear cache of expired item
+                $this->deleteItem($item->getKey());
+            }
+            unset($this->deferredItems[$key]);
         }
+        $this->deferredItems = [];
+
+        return $this->cache->storeToPool($deferred, $this->poolName);
```

Comment:
not ok but we say it is for sake of pushing forward
@llegaz investigate possible implementation using Valkey.io